### PR TITLE
[UI] Added Dark/Light mode 

### PR DESCRIPTION
**Pull request by maikyh**
## Description
- This PRs adde a button that enables users to switch between light and dark modes for the website's appearance.

## Milestones
- User can see the website in light/dark mode according to their preferences

## Test Plan
New Note
New Note
New Note
New Note
New Note
New Note
New Note

### CODE REVIEW

File: ans2ask-ui/src/App.jsx [hunk 1]
```diff
@@ -35,6 +36,22 @@ export default function App() {
     localStorage.setItem('user', JSON.stringify(user));
   }, [user]);
 
+  //dark mode
+  const [darkMode, setDarkMode] = useState(() => {
+    // Retrieve the user data from storage or set it to null if not found
+    const storedDarkMode = localStorage.getItem('darkMode');
+    return storedDarkMode ? JSON.parse(storedDarkMode) : true;
+  });
```
Comment (chengkaoma) [inline ans2ask-ui/src/App.jsx:44]: Is this used to improve the performance?

File: ans2ask-ui/src/App.jsx [hunk 2]
```diff
@@ -66,10 +83,19 @@ export default function App() {
     return () => clearTimeout(timer);
   }, [images])
 
+  useEffect(() => {
+    if(darkMode){
+      document.body.style.backgroundColor = '#1A202C';
```
Comment (chengkaoma) [inline ans2ask-ui/src/App.jsx:88]: make hard-code background color to a constant in the class or use enums

**Review by chengkaoma — COMMENTED** [review]
A lot of colors look like duplicated. 
Is it possible to put them into utils and reuse on all places?

**Review by chengkaoma — COMMENTED** [review]
Also is this PR mixed with previous PR? 
I see a lot of changes related to search algorithm

File: ans2ask-ui/src/App.jsx (hunk 1 above)
Comment (maikyh) [reply ans2ask-ui/src/App.jsx:44]: this is to get the user's stored preference, if it is the first time to visit the website, the default mode will be dark

**Comment by maikyh** [comment]
> Also is this PR mixed with previous PR? I see a lot of changes related to search algorithm

Yes, i created a branch out of the previous branch, that's why it shows some search algo related stuff

**Pull request merged**